### PR TITLE
intro-to-solana-mobile.md - refactor: optimize counter increment logic and notification handling

### PR DESCRIPTION
**Pull request by EmekaManuel**
### Problem
- The showToastOrAlert function name was unclear, and error handling in async functions was inconsistent.
- The logic for balance checks and airdrop requests during transactions needed improvements.
- Detailed error logging and cleanup were required for better readability and maintainability.

### Summary of Changes
Refactored showToastOrAlert to showNotification for better clarity.
Improved async/await usage with consistent error handling using try/catch.
Added detailed logging for missing program or counterAddress.
Cleaned up and grouped logic for better readability and maintainability.
Ensured balance checks and airdrop requests are well-handled during transactions.

### CODE REVIEW

**Comment by mikemaccana** [comment]
Thanks! Needs a little work but we can get this in once done.

**Comment by EmekaManuel** [comment]
> Thanks! Needs a little work but we can get this in once done.

Thank you Mike! Looking forward to that.

**Review by mikemaccana — CHANGES_REQUESTED** [review]
Some small changes needed but also not many.

File: content/courses/mobile/intro-to-solana-mobile.md [hunk 1]
```diff
@@ -1008,11 +1008,27 @@ export function CounterView() {
   const { program, counterAddress } = useProgram();
   const [counter, setCounter] = useState<CounterAccount>();
 
-  // Fetch Counter Info
   useEffect(() => {
-    if (!program || !counterAddress) return;
+    if (!program || !counterAddress) {
+      console.log("Missing dependencies:", {
+        program: !!program ? "Program Loaded" : "Program Not loaded",
```
[the remark this answers is not in the document]
Comment (EmekaManuel) [reply content/courses/mobile/intro-to-solana-mobile.md:1014]: fixed !

File: content/courses/mobile/intro-to-solana-mobile.md [hunk 2]
```diff
@@ -1109,62 +1125,63 @@ export function CounterButton() {
...
-  const showToastOrAlert = (message: string) => {
-    if (Platform.OS === "android") {
-      ToastAndroid.show(message, ToastAndroid.SHORT);
-    } else {
-      Alert.alert(message);
-    }
+  const showNotification = (message: string) => {
+    Platform.OS === "android"
+      ? ToastAndroid.show(message, ToastAndroid.SHORT)
+      : Alert.alert(message);
```
[the remark this answers is not in the document]
Comment (EmekaManuel) [reply content/courses/mobile/intro-to-solana-mobile.md:1131]: fixed !

File: content/courses/mobile/intro-to-solana-mobile.md [hunk 3]
```diff
@@ -1109,62 +1125,63 @@ export function CounterButton() {
   const { connection } = useConnection();
   const [isTransactionInProgress, setIsTransactionInProgress] = useState(false);
 
-  const showToastOrAlert = (message: string) => {
-    if (Platform.OS === "android") {
-      ToastAndroid.show(message, ToastAndroid.SHORT);
-    } else {
-      Alert.alert(message);
-    }
+  const showNotification = (message: string) => {
```
[the remark this answers is not in the document]
Comment (EmekaManuel) [reply content/courses/mobile/intro-to-solana-mobile.md:1128]: this does show notification. can you please elaborate on this review

[the remark this answers is not in the document]
Comment (mikemaccana) [reply content/courses/mobile/intro-to-solana-mobile.md:1129]: Does it create one of these?

image

**Review by mikemaccana — CHANGES_REQUESTED** [review]
Just one last bit and we can get this in.

File: content/courses/mobile/intro-to-solana-mobile.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (EmekaManuel) [reply content/courses/mobile/intro-to-solana-mobile.md:1129]: It creates these instead: 
alert_example_ios
alert_example_android

[the remark this answers is not in the document]
Comment (mikemaccana) [reply content/courses/mobile/intro-to-solana-mobile.md:1129]: That's not a 'notification' then. The word notification has a very specific meaning on mobile. Please change the function name.

**Review by mikemaccana — CHANGES_REQUESTED** [review]
Just need to fix function name per above. `showMessage()` would be fine. Just not 'notification'.

File: content/courses/mobile/intro-to-solana-mobile.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (EmekaManuel) [reply content/courses/mobile/intro-to-solana-mobile.md]: changed to showMessage()

**Comment by EmekaManuel** [comment]
> Just need to fix function name per above. `showMessage()` would be fine. Just not 'notification'.

changed to showMessage✅

**Pull request closed without merging**